pkgs/desktops/gnome-3/3.20/core/gnome-shell/default.nix

```diff
@@ -1,9 +1,10 @@
 { fetchurl, stdenv, pkgconfig, gnome3, json_glib, libcroco, intltool, libsecret
-, python3, libsoup, polkit, clutter, networkmanager, docbook_xsl, docbook_xsl_ns, at_spi2_core
-, libstartup_notification, telepathy_glib, telepathy_logger, libXtst, p11_kit, unzip
-, sqlite, libgweather, libcanberra_gtk3
-, libpulseaudio, libical, libtool, nss, gobjectIntrospection, gstreamer, makeWrapper
-, accountsservice, gdk_pixbuf, gdm, upower, ibus, networkmanagerapplet, librsvg }:
+, python3, python3Packages, libsoup, polkit, clutter, networkmanager
```

Comment:
no need to have `python3` and `python3Packages` since you can use `python3Packages.python` to obtain the interpreter. Furthermore, the interpreter is a runtime dependency of the Python packages.